src/viam/sdk/common/utils.hpp

```diff
@@ -44,5 +44,8 @@ class ResourceNameEqual {
 std::vector<unsigned char> string_to_bytes(std::string const& s);
 std::string bytes_to_string(std::vector<unsigned char> const& b);
 
+std::chrono::duration<double> duration_from_proto(const google::protobuf::Duration& proto);
+google::protobuf::Duration duration_to_proto(const std::chrono::duration<double>& duration);
```

Comment:
How do we feel about `std::chrono::duration<double>` as the default viam::sdk duration unit? Some online research seems to say that this is the approved and portable way of doing things (and it lets people use whatever units they want).

I was thinking of `typedef`ing this like elsewhere but I am worried that it will make the type even harder to grok

Reply:
I wouldn't typedef it. I think `std::chrono::duration<double>` makes it very clear what it is. I need to think about how I feel about using `double` with `std::chrono::duratuion` a bit. It was surprising to me at first but I think I'm seeing why it might be a good choice.

Reply:
@zaporter-work - I think it would be helpful if you outlined your argument for using `double` as the type for `Rep` in `duration`.

Reply:
I did this for two reasons: 
1)
We are given a `int64_t` seconds and a `int32_t` from proto duration. 
To the best of my understanding, there is no `int128_t` in C++ (even `long long` is 64b).
So if we want to store everything under a single number, we must use a 64 bit type.

I had thought about using `duration<long long, nanoseconds>` but then I am sacrificing precision in the important dimension (seconds) for a bit of extra precision in the less important dimension (nanoseconds).

This led me towards creating a `duration<double, seconds>` from the proto generated code.

This has the risk of losing precision in the nanoseconds dimension (however we have a test that tests no loss of precision at least for small numbers).

2) 
It makes it slightly easier to convert between different time units in an agnostic way (ex: if a user casts our `seconds`-based duration to `minutes` they will have to intentionally chose to lose precision. If it is a double, then no information will be rounded or truncated)

Reply:
I do not have deep experience with this type though, so if either of you have strong opinions, I am happy to go with `std::chrono::duration<long long>`

Reply:
So, interestingly, the 64-bit seconds part of a `proto.Duration` has a limited range, smaller than the full signed 64-bit range: https://protobuf.dev/reference/protobuf/google.protobuf/#duration.

That range happens to be an interesting looking number, so I went looking for a justification, and found one: https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/duration.proto#L103.

We cannot represent +/- 10ky in int64_t nanos: (2^63) / (10,000 * 365.25 * 24 * 60 * 60 * 1,000,000,000) = 0.0292271023...

But we **can** represent +/- 10ky in int64_t micros: (2^63) / (10,000 * 365.25 * 24 * 60 * 60 * 1,000,000) = 29.227...

We do lose a tiny bit of precision, but at most one microsecond. So I'd be pretty comfortable declaring that our vocabulary duration type in the Viam C++ SDK is denominated in microseconds.

If we are uncomfortable with loosing a whole microsecond, it looks to me like we have another decimal place, so if we denominated in units of 100 nanoseconds that would be our maximum precision loss when going to/from proto. It is a non-standard chrono ratio though.

Reply:
I'm a fan of representing time with int over float and I don't expect that nano precision is going to be especially important here, so I'm in favor of using micro denomination with int64_t. The potential loss of a microsecond is a bit sad but I don't think it'll be terrible in practice, and is something that we can easily account for in tests as necessary.

Reply:
I have just made this change in c89803d4cf8db3ffa1389573a79f7e96e6870752. I think it is slightly less ergonomic but it is nice to have integer-based durations.

Reply:
@stuqdog @acmorrow If you could skim that commit and confirm that is what is preferred, that would be appreciated.